src/search.rs

```diff
@@ -31,11 +40,42 @@ pub async fn find(req: HttpRequest) -> HttpResponse {
 		param(&path, "sort")
 	};
 	let sub = req.match_info().get("sub").unwrap_or("").to_string();
+	let mut subreddits: Vec<Subreddit> = Vec::new();
+	
+	if param(&path, "restrict_sr") == "" {
+		let subreddit_search_path = format!("/subreddits/search.json?q={}&limit=3", param(&path, "q"));
+		let res;
+		let subreddit_list;
+
+		// Send a request to the url
+		match request(&subreddit_search_path).await {
+			// If success, receive JSON in response
+			Ok(response) => {
+				res = response;
+				subreddit_list = res["data"]["children"].as_array();
+			}
+			// If the Reddit API returns an error, exit this function
+			Err(_msg) => {subreddit_list = None;}
+		}
```

Comment:
Can this request be made parallel with the main search request?

Reply:
Not sure but I plan to do add concurrency project-wide. I can merge this and then take a look at it.